doc/tutorials/using_ubuntu_as_sos.rst

```diff
@@ -0,0 +1,262 @@
+      sudo apt-get install rpm2cpio
+      rpm2cpio linux-pk414-sos-4.14.41-39.x86_64.rpm | cpio -idmv
+
+#. Install the SOS kernel and its drivers (modules)
+
+   .. code-block:: console
+
+      sudo cp -r ~/kernel-build/usr/lib/modules/4.14.41-39.pk414-sos/ /lib/modules/
+      sudo cp ~/kernel-build/usr/lib/kernel/org.clearlinux.pk414-sos.4.14.41-39 /boot/acrn/
+
+#. Configure Grub to load the Service OS kernel
+
+   * Modify the ``/etc/grub.d/40_custom`` file to create a new Grub entry that will boot the SOS kernel.
+
+     .. code-block:: none
+
+        menuentry 'ACRN ubuntu SOS' {
+                recordfail
+                load_video
+                insmod gzio
+                insmod part_gpt
+                insmod ext4
+                linux /boot/acrn/org.clearlinux.pk414-sos.4.14.41-39 pci_devices_ignore=(0:18:1) maxcpus=1 console=tty0 console=ttyS0
+                i915.nuclear_pageflip=1 root=PARTUUID=<ID of rootfs partition> rw rootwait ignore_loglevel no_timer_check consoleblank=0
+                i915.tsd_init=7 i915.tsd_delay=2000 i915.avail_planes_per_pipe=0x00000F i915.domain_plane_owners=0x011111110000
+                i915.enable_guc_loading=0 i915.enable_guc_submission=0 i915.enable_preemption=1 i915.context_priority_mode=2
+                i915.enable_gvt=1 hvlog=2M@0x1FE00000
+        }
+
+     .. note::
+        You need to adjust this to use your partition UUID (``PARTUUID``) for the ``root=`` parameter (or use the device node directly).
+
+     .. note::
+        You will also need to adjust the kernel name if you used a different RPM file as the source of your Service OS kernel.
+
+   * Update Grub on your system
+
+     .. code-block:: console
+
+        sudo update-grub
+
+     At this point, you need to modify ``/boot/grub/grub.cfg`` file manually to enable the timeout so that the system has opportunity to show
+     you grub menu to select. Or you need to set the “ACRN ubuntu SOS” as the default boot option. Without enabling timeout, the system
+     will not show grub menu for people to select and boot into incorrect kernel.
+
+     .. code-block:: none
+
+          if [ x$feature_timeout_style = xy ] ; then
+        -  set timeout_style=hidden
+        + #set timeout_style=hidden
+        -  set timeout=0
+        + set timeout = 10
+          # Fallback hidden-timeout code in case the timeout_style feature is
+          # unavailable.
+
+#. Reboot the system
+   
+   Reboot system. You should see the Grub menu with the new “ACRN ubuntu SOS” entry. Select it and proceed to booting the platform.
+   The system will start the Ubuntu Desktop and you can now log in (as before).
+
+   To check if the hypervisor is effectively running, check ``dmesg``. The typical output of a successful installation
+   will look like this:
+
+   .. code-block:: none
+
+      dmesg | grep ACRN
+      [    0.000000] Hypervisor detected: ACRN
+      [    0.862942] ACRN HVLog: acrn_hvlog_init
+
+Prepare the User OS (UOS)
```

Comment:
If this isn't any different than the getting started guide, I'd delete this part and refer them to that guide to finish up with the User OS.  (Duplicating instructions is a bad idea.)

Reply:
There are a few differences:
* `chmod` of the device node
* nstallation of the `iasl` package
* Downloading the PK (standard) kernel

And the usual adjustment to the clear version (and kernel version) in the scripts (but that is covered in the text).

Reply:
In that case, maybe add a mention that these instructions are slightly different than what in the getting-started-apl-nuc doc.  Are the differences because we're using Ubuntu as the SOS, if not then why are they different?